### PR TITLE
Add Collection object

### PR DESCRIPTION
**Pull request by squirrelo**
This fleshes out the collection object so it can be used in further pull requests.

### CODE REVIEW

File: qiita_db/analysis.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (ElDeveloper) [inline qiita_db/analysis.py]: Can you remove this extra whitespace?

File: qiita_db/analysis.py [hunk 2]
```diff
@@ -806,41 +818,90 @@ def create(cls, owner, name, description=None):
...
+               "VALUES (%s, %s, %s)".format(cls._table))
+        conn_handler.execute(sql, [owner.id, name, description])
 
     @classmethod
     def delete(cls, id_):
         """Deletes a collection from the database"""
-        raise NotImplementedError()
+        conn_handler = SQLConnectionHandler()
+        if cls(id_).status == "public":
+            raise QiitaDBStatusError("Can't delete public study!")
```
Comment (ElDeveloper) [inline qiita_db/analysis.py]: Is this message accurate? Seems like you are removing a study collection not a study.

File: qiita_db/analysis.py [hunk 3]
```diff
@@ -806,41 +818,90 @@ def create(cls, owner, name, description=None):
...
+
+        sql = ("INSERT INTO qiita.{0} (email, name, description) "
+               "VALUES (%s, %s, %s)".format(cls._table))
+        conn_handler.execute(sql, [owner.id, name, description])
 
     @classmethod
     def delete(cls, id_):
         """Deletes a collection from the database"""
-        raise NotImplementedError()
+        conn_handler = SQLConnectionHandler()
```
Comment (ElDeveloper) [inline qiita_db/analysis.py]: It seems like the docstring of this method is missing the description of id_.

Comment (ElDeveloper) [inline qiita_db/analysis.py]: Also, it would be good to add a Raises section mentioning QiitaDBStatusError.

**Comment by ElDeveloper** [comment]
This looks fine, though I don't have a compelling use-case to try this object out.

File: qiita_db/analysis.py (hunk 1 above)
Comment (squirrelo) [inline qiita_db/analysis.py]: Not extra, pep8 requires it.

**Comment by squirrelo** [comment]
Ready to go.

File: qiita_db/analysis.py (hunk 1 above)
Comment (ElDeveloper) [inline qiita_db/analysis.py]: That's a new one ... but I guess if PEP8 says so.

**Comment by ElDeveloper** [comment]
:+1:

**Comment by ElDeveloper** [comment]
If someone else can review, this should be ready to be merged.

**Pull request merged**